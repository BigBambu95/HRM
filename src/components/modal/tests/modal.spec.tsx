import React from 'react'
import Modal, { ModalProps } from '../modal'
import { shallow } from 'enzyme'

describe('Modal', () => {
	const setUp = (props?: ModalProps, children?: React.ReactNode) =>
		shallow(
			<Modal isOpen={false} onSubmit={() => {}} onCancel={() => {}} {...props}>
				{children}
			</Modal>
		)

	it('Проверка что модальное окно не отрендерено по умолчанию', () => {
		const modal = setUp()
		expect(modal).toMatchSnapshot()
	})

	it('Рендер окна с пропсами по умолчанию', () => {
		const modal = setUp({ isOpen: true })
		expect(modal).toMatchSnapshot()
	})

	it('Рендер окна с переданными пропсами', () => {
		const modal = setUp(
			{
				isOpen: true,
				title: 'Создать вакансию',
				submitBtnText: 'Создать',
				width: 800,
			},
			<div>Custom children</div>
		)

		expect(modal).toMatchSnapshot()
	})
})
