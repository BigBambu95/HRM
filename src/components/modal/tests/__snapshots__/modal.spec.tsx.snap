// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Проверка что модальное окно не отрендерено по умолчанию 1`] = `""`;

exports[`Modal Рендер окна с переданными пропсами 1`] = `
<div
  className="modal-window-wrapper"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="modal-window"
    style={
      Object {
        "maxWidth": 800,
      }
    }
  >
    <div
      className="modal-window__header"
    >
      <h2>
        Создать вакансию
      </h2>
      <Button
        className="modal-window__close-btn"
        onClick={[Function]}
        variant="icon"
      >
        <CloseIcon />
      </Button>
    </div>
    <div
      className="modal-window__body"
    >
      <div>
        Custom children
      </div>
    </div>
    <div
      className="modal-window__footer"
    >
      <Button
        font="large"
        onClick={[Function]}
        size="large"
        variant="solid"
      >
        Создать
      </Button>
      <Button
        onClick={[Function]}
        size="large"
      >
        Отменить
      </Button>
    </div>
  </div>
</div>
`;

exports[`Modal Рендер окна с пропсами по умолчанию 1`] = `
<div
  className="modal-window-wrapper"
>
  <div
    className="overlay"
    onClick={[Function]}
  />
  <div
    className="modal-window"
    style={
      Object {
        "maxWidth": 600,
      }
    }
  >
    <div
      className="modal-window__header"
    >
      <h2>
        Сформировать приказ
      </h2>
      <Button
        className="modal-window__close-btn"
        onClick={[Function]}
        variant="icon"
      >
        <CloseIcon />
      </Button>
    </div>
    <div
      className="modal-window__body"
    />
    <div
      className="modal-window__footer"
    >
      <Button
        font="large"
        onClick={[Function]}
        size="large"
        variant="solid"
      >
        Отправить
      </Button>
      <Button
        onClick={[Function]}
        size="large"
      >
        Отменить
      </Button>
    </div>
  </div>
</div>
`;
